### ISSUE TITLE
Get thumbnail from cache?

### ISSUE THREAD

**Issue by eneim** [danikula/AndroidVideoCache#33]
Could this library help to get thumbnail from videos **WHILE** loading them?

**Comment by danikula** [comment]
@eneim I think it is possible to take screenshot for not fully downloaded video. You can try to use [this](http://developer.android.com/reference/android/media/ThumbnailUtils.html#createVideoThumbnail(java.lang.String, int)) or [this](http://developer.android.com/reference/android/media/MediaMetadataRetriever.html#getFrameAtTime()) code for retrieving bitmap during downloading. Just set `CacheListener` for proxy, wait for downloading part of video and try to take screenshot. Anyway this improvement is out of scope of library's goals and will not be implemented. You can fork the project and implement by yourself.

**Comment by eneim** [comment]
Thanks for your response. In case ThumbnailUtil doesn't work (which I suppose to support only fully stored video file), I will try to implement it separately

**Comment by danikula** [comment]
@eneim Please, let me know if you will implement it successfully

**Comment by eneim** [comment]
@danikula I made a quick test and the following snippet seems to work

```
boolean isThumbnailExtracted = false;
if (percentsAvailable >= 10 && !isThumbnailExtracted) { // extract only after we have something, and once
            Bitmap thumb = ThumbnailUtils.createVideoThumbnail(file.getPath(),
                MediaStore.Images.Thumbnails.MINI_KIND);
            if (thumb != null) {
                isThumbnailExtracted = true; // done, don't need to do more
                imageView.setImageBitmap(thumb); // setup some imageView here
            }
        }
```

For production usage, it might need some improvement, but it works nicely now.

**Comment by danikula** [comment]
@eneim nice
